src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -0,0 +1,548 @@
+                            projectPath));
+
+                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    continue;
+                }
+
+                var lockFileFormat = new LockFileFormat();
+                LockFile assetsFile = lockFileFormat.Read(assetsPath);
+
+                // assets file validation
+                if (assetsFile.PackageSpec == null
+                    || assetsFile.Targets == null
+                    || assetsFile.Targets.Count == 0)
+                {
+                    Console.Error.WriteLine(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.WhyCommand_Error_CannotReadAssetsFile,
+                            assetsPath));
+
+                    ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+                    continue;
+                }
+
+                // get all resolved package references for a project
+                List<FrameworkPackages> packages = msBuild.GetResolvedVersions(project, whyCommandArgs.Frameworks, assetsFile, transitive: true);
+
+                if (packages?.Count > 0)
+                {
+                    FindAllDependencyGraphs(packages, assetsFile.Targets, targetPackage, projectName);
+                }
+                else
+                {
+                    Console.WriteLine(
+                        string.Format(
+                            Strings.WhyCommand_Message_NoPackagesFoundForGivenFrameworks,
+                            projectName));
+                }
+
+                // unload project
+                ProjectCollection.GlobalProjectCollection.UnloadProject(project);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Runs the 'why' command, and prints output to the console.
+        /// </summary>
+        /// <param name="frameworkPackages">All frameworks in the project, with their top-level packages and transitive packages.</param>
+        /// <param name="targets">All lock file targets in the project.</param>
+        /// <param name="targetPackage">The package we want the dependency paths for.</param>
+        /// <param name="projectName">The name of the current project.</param>
+        private void FindAllDependencyGraphs(
+            IEnumerable<FrameworkPackages> frameworkPackages,
+            IList<LockFileTarget> targets,
+            string targetPackage,
+            string projectName)
+        {
+            bool doesProjectHaveDependencyOnPackage = false;
+            var dependencyGraphPerFramework = new Dictionary<string, List<DependencyNode>>(targets.Count);
+
+            foreach (var frameworkPackage in frameworkPackages)
+            {
+                LockFileTarget target = targets.FirstOrDefault(f => f.TargetFramework.GetShortFolderName() == frameworkPackage.Framework);
```

Comment:
We should be using strings instead of NugetFramework here right? 

We should make sure that why works with maui projects, which is where you'd see these challenges. 

We can look together if you need help around this topic.

Reply:
I think I linked a method up above that might be the one we need to use. GetLockFileTarget or something like that. 

using that might do enough.

Reply:
I'm now using the GetTarget() method that you referred to above.

The code here also uses the `GetResolvedVersions()` method from MSBuildAPIUtility, and that extracts information from the assets file's targets based on user input framework strings. This code might need changes too, but I wasn't sure how to implement them: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs#L686-L712

I'll reach out to you offline to see what changes this might need.